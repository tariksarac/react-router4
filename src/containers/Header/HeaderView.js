import React, { Component } from 'react';
import { Link } from 'react-router-dom';

import './Header.css'

class Header extends Component {

    redirectToHomepage(){
        this.props.history.push('/')
    }
    render() {
        return (
            <header>
                <div onClick={this.redirectToHomepage.bind(this)}>
                    ReticleAppCenter
                </div>
                <nav>
                    {
                        this.props.authenticated ?
                            <ul>
                                <li>
                                    <Link to="/">Home</Link>
                                </li>
                                <li>
                                    <Link to="auth/logout">Logout</Link>
                                </li>
                                <li>
                                    <Link to="react-grid">Grid</Link>
                                </li>
                            </ul>
                            :
                            <ul>
                                <li>
                                    <Link to="/auth">Login</Link>
                                </li>
                                <li>
                                    <Link to="/auth/register">Register</Link>
                                </li>
                            </ul>
                    }
                </nav>
            </header>
        )
    }
}

export default Header;