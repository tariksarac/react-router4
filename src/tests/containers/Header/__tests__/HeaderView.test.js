import React from 'react'
import HeaderView from '../../../../containers/Header/HeaderView'
import { shallow } from 'enzyme'


it('should render HeaderView without crashing', () => {
    const app = shallow(<HeaderView />);
    expect(app).toMatchSnapshot()
});