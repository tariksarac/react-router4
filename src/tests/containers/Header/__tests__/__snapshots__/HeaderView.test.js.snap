// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render HeaderView without crashing 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <header>
    <div
        onClick={[Function]}
    >
        ReticleAppCenter
    </div>
    <nav>
        <ul>
            <li>
                <Link
                    replace={false}
                    to="/auth"
                >
                    Login
                </Link>
            </li>
            <li>
                <Link
                    replace={false}
                    to="/auth/register"
                >
                    Register
                </Link>
            </li>
        </ul>
    </nav>
</header>,
  "nodes": Array [
    <header>
      <div
            onClick={[Function]}
      >
            ReticleAppCenter
      </div>
      <nav>
            <ul>
                  <li>
                        <Link
                              replace={false}
                              to="/auth"
                        >
                              Login
                        </Link>
                  </li>
                  <li>
                        <Link
                              replace={false}
                              to="/auth/register"
                        >
                              Register
                        </Link>
                  </li>
            </ul>
      </nav>
</header>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <Header />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": Header {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <header>
          <div
                    onClick={[Function]}
          >
                    ReticleAppCenter
          </div>
          <nav>
                    <ul>
                              <li>
                                        <Link
                                                  replace={false}
                                                  to="/auth"
                                        >
                                                  Login
                                        </Link>
                              </li>
                              <li>
                                        <Link
                                                  replace={false}
                                                  to="/auth/register"
                                        >
                                                  Register
                                        </Link>
                              </li>
                    </ul>
          </nav>
</header>,
        "_debugID": 2,
        "_renderedOutput": <header>
          <div
                    onClick={[Function]}
          >
                    ReticleAppCenter
          </div>
          <nav>
                    <ul>
                              <li>
                                        <Link
                                                  replace={false}
                                                  to="/auth"
                                        >
                                                  Login
                                        </Link>
                              </li>
                              <li>
                                        <Link
                                                  replace={false}
                                                  to="/auth/register"
                                        >
                                                  Register
                                        </Link>
                              </li>
                    </ul>
          </nav>
</header>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Header />,
}
`;
